R/utils.R

```diff
@@ -34,9 +34,11 @@ install_saspy <- function(method = "auto", conda = "auto") {
 #'
 #' @keywords internal
 validate_ssh_with_tunnel <- function(session, msg = "SAS session through ssh must use tunnels to transfer datasets!") {
-  cfgname <- session$sascfg$SAScfg$SAS_config_names[1]
+  cfgname <- session$sascfg$name
+  is_ssh <- identical(session$sascfg$mode, "ssh")
```

Comment:
I think identical() will only be TRUE if you match case. So use "SSH" for the literal, as the mode attribute is upper cased in saspy. 
Out of curiosity, why require tunneling with the SSH access method? It seems that any connection that's established via SASPy works, so why require this in the first place? I don't have an R install that works with this, so I haven't been able to try it out to see if I can answer that question myself, but it seems that this shouldn't care how saspy is connected; just that it is.
Thanks,
Tom

Reply:
thank you Tom. indeed this is upper case. the reason I add it is that a while ago I experienced issues transfering data for ssh without tunneling, but it seems it is working as expected now.
thank you so much!